### PR TITLE
feat(http, coordinator): explainPlan support in FiloDB HTTP

### PR DESCRIPTION
**Pull request by TanviBhavsar**
**Pull Request checklist**

- [x] The commit(s) message(s) follows the contribution [guidelines](CONTRIBUTING.md) ?
- [x] Tests for the changes have been added (for bug fixes / features) ?
- [x] Docs have been added / updated (for bug fixes / features) ?

**New behavior :**
 Adds ability to get ExecPlan from FiloDB HTTP

### CODE REVIEW

**Comment by TanviBhavsar** [comment]
> 👍
> 
> Can you document in your PR how you tested this change? Was it local E2E test?

Yes I tested it locally using explainPlan = true

**Comment by velvia** [comment]
Oh @TanviBhavsar  it would be great to have a unit test as well.

**Comment by vishramachandran** [comment]
@velvia The reason it was not a separate route was because we'd have one route for each type of query
* query
* query_range
* metadata
etc.

One could argue that the response schema is not vastly different. It just has an additional field "debuginfo". We could also aim for printing the plan along with the result in the future.

**Comment by velvia** [comment]
@TanviBhavsar ok if we go with param instead of separate route, then please rename the param to `explainOnly` as this is more accurate - implies there is no query

**Comment by TanviBhavsar** [comment]
> @TanviBhavsar ok if we go with param instead of separate route, then please rename the param to `explainOnly` as this is more accurate - implies there is no query



> @TanviBhavsar ok if we go with param instead of separate route, then please rename the param to `explainOnly` as this is more accurate - implies there is no query

@velvia Changed explainPlan to explainOnly. Also updated doc/http_api.md

File: query/src/main/scala/filodb/query/exec/ExecPlan.scala [hunk 1]
```diff
@@ -198,6 +198,16 @@ trait ExecPlan extends QueryCommand {
     ((transf :+ curNode) ++ childr).mkString(if (useNewline) "\n" else " @@@ ")
   }
 
+  final def getPlan(level: Int = 0): Seq[String] = {
+    val transf = rangeVectorTransformers.reverse.zipWithIndex.map { case (t, i) =>
+      s"${"-"*(level + i)}T~${t.getClass.getSimpleName}(${t.args})"
```
Comment (velvia) [inline query/src/main/scala/filodb/query/exec/ExecPlan.scala]: Might want to put this part which is shared in a function

**Comment by velvia** [comment]
@TanviBhavsar good work and thanks for the test and changes!   Will approve after explain param is changed back to explainOnly .... not sure if it was accidentally changed back in the last commit.

File: query/src/main/scala/filodb/query/exec/ExecPlan.scala (hunk 1 above)
Comment (TanviBhavsar) [reply query/src/main/scala/filodb/query/exec/ExecPlan.scala]: Missed this before. Added function to print RangeVectorTransformers

**Comment by TanviBhavsar** [comment]
> @TanviBhavsar good work and thanks for the test and changes! Will approve after explain param is changed back to explainOnly .... not sure if it was accidentally changed back in the last commit.

Thanks Evan! I have made the changes

**Comment by velvia** [comment]
@TanviBhavsar thanks, I already approved but one more request ...

The http_api.md doc changes imply only the `query_range` route supports the param, but query also right?  Instead of creating a separate entry, I'd add a note to both the query and query_range sections that the param `explainOnly` returns exec plan only.  Thanks

**Comment by TanviBhavsar** [comment]
> @TanviBhavsar thanks, I already approved but one more request ...
> 
> The http_api.md doc changes imply only the `query_range` route supports the param, but query also right? Instead of creating a separate entry, I'd add a note to both the query and query_range sections that the param `explainOnly` returns exec plan only. Thanks

Removed separate entry and added param to  query and query_range section

**Pull request merged**